addons/website/static/src/js/content/cookie_watcher.js

```diff
@@ -0,0 +1,46 @@
+// This script is to be called directly in the <head>. It should not import any
+// other file or library, as it cannot be transformed into a module (because a
+// module is deferred): the function should be the very first thing launched on
+// the page to be able to bypass other IIFE of 3rd-party services that inject
+// new scripts client-side.
+
+// eslint-disable-next-line no-unused-vars
+function watch3rdPartyScripts(thirdPartyDomainsBlockList) {
+    const blockList = thirdPartyDomainsBlockList.join("|").replaceAll(".", "\\.");
+    // Any of "www.domain.com", "domain.com" and "domain" should match
+    // "www.domain.com" (but not "www.mydomain.com").
+    const hostRegex = new RegExp(`^(.*\\.)?(${blockList})(\\..*)?$`, "i");
+    const cookieRegex = /(^|(; ))website_cookies_bar=(?<value>[^;]+)/;
+    const scriptSrcDesc = Object.getOwnPropertyDescriptor(HTMLScriptElement.prototype, "src");
+    Object.defineProperty(HTMLScriptElement.prototype, "_src", scriptSrcDesc);
+    Object.defineProperty(HTMLScriptElement.prototype, "src", {
```

Comment:
In fact why do we do this for script but not iframe?

Reply:
Overriding the prototype is quite invasive, and while that was the only solution to prevent scripts generated by IIFEs (mainly analytics/tracking scripts) from launching, iframes could be done in several ways.

To me, the 2 main differences are:
- iframes injected by the customer (if any) would not generally be constructed client-side: they are saved in the view, and the iframes' src are checked on page generation in python. That leaves us with a few iframes that website builds client-side through the widgets (instagram, maps, etc.): a limited list that we can manage directly.
- we "replace" the iframe with a warning: it must be translated (so can't happen before the `assets_frontend` bundle), and the iframe needs a parent (the warning is added as a sibling). If the `src` is set through JS, it generally means that the iframe is being built and is not in the DOM yet: it probably doesn't have a parent yet. That quickly gets overly complicated for nothing in my opinion, and that doesn't prevent developers from having to think about it in future implementations (if anything, it's less obvious that they should pay attention to it I think, because it's more conceiled).

To this, you also have to take into account that:
- as said, overriding the prototype is invasive
- it only works if the src is set directly (as `el.src = ''`, not with `el.setAttribute('src', ...)`, nor with jQuery's `$('iframe', { src })`). From what I see, mostly not a problem for scripts, but could be for iframes. And overriding `setAttribute` on the prototype doesn't look like a good idea.
- we would still need to handle iframes saved in the view anyway

Overall, I believe it would end up with a needlessly convoluted code, more prone to bugs, for little benefit.